torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -89,24 +32,111 @@ def _get_params(self, sample: Any) -> Dict[str, Any]:
         elif fill is not None:
             fill = [float(f) for f in fill]
 
-        return dict(interpolation=self.interpolation, fill=fill)
+        return dict(fill=fill)
 
     def _get_random_item(self, dct: Dict[K, V]) -> Tuple[K, V]:
         keys = tuple(dct.keys())
         key = keys[int(torch.randint(len(keys), ()))]
         return key, dct[key]
 
-    def _apply_transform(self, sample: Any, params: Dict[str, Any], transform_id: str, magnitude: float) -> Any:
-        dispatcher = self._DISPATCHER_MAP[transform_id]
-
-        def transform(input: Any) -> Any:
-            if type(input) in {features.Image, torch.Tensor} or isinstance(input, PIL.Image.Image):
-                return dispatcher(input, magnitude, params["interpolation"], params["fill"])
+    def _apply_transform(self, sample: Any, transform_id: str, magnitude: float) -> Any:
+        def dispatch(image_kernel: Callable, legacy_kernel: Callable, input: Any, *args: Any, **kwargs: Any) -> Any:
```

Comment:
Any chance we can simplify this?

Reply:
Could you elaborate what you want to be simplified? I don't think there is room for significant simplification without introducing more abstractions.

Reply:
Let's have a chat offline to see pros/cons.